### ISSUE TITLE
HowTo/readme

### ISSUE THREAD

**Issue by DutchmanNL** [energychain/Corrently-Node-Discovergy#1]
would it be possible to add how to use this npm source te retrieve data from the API and which code samples should be used ?

**Comment by zoernert** [comment]
This module is part of the Corrently EcoSystem. So an isolated example to retrieve data from Discovergy API would not make too much sense. In General this module uses the Discovergy API as described at https://api.discovergy.com/docs/ . Username and password is set via environment variables (.env file).

Hope that helps?

**Comment by DutchmanNL** [comment]
hi @zoernert thank you very much for your reply.

Ehm, no still struggeling to be honest.
I am trying to find a find reading the data with javascript formatting and thought using this NPM pakkage to handle the auhtentication part would be quite usefull.

The API documentation does not help me much, maybe i am to simple, basically i would like to undestand how i could provide the username/pass to the API/NPM package and get in return the raw JSON file provided by dicovergy.

Backgound information : I would like to include the discovergy information into the iObroker iOT automation system.
We have already quite some nice adapters by discovergy is missing.
I can use javascript and install the NPM module to be included into the iObroker javascript engine, but the the question starts how to handle to authentication and retrieve data.

i would be happy to just get the raw JSON as logik for reading the elements would be handled by the iOB software

**Comment by zoernert** [comment]
As I was just recently looking at the IOBroker  GIT ( https://github.com/ioBroker/ ) you got my attention :)

So long story short. Start with the following snippet (which is actually a simple HTTPs Request to the API). If you put it into an adapter - share the repository with me as a contributor, to make it nice :)

```
const http_request = require("request");

function doDiscovergyCall(username, password,endpoint,urlencoded_parameters) {
  return new Promise(function (resolve, reject)  {
    http_request("https://"+username+":"+password+"@api.discovergy.com/public/v1/"+endpoint+"?"+urlencoded_parameters,function(r,e,b) {
      console.log(b);
      resolve(JSON.parse(b));
    });
  });
}

doDiscovergyCall("demo@discovergy.com","demo","meters","").then(function(meters) {
  console.log(meters);
})

doDiscovergyCall("demo@discovergy.com","demo","last_reading","meterId=0272eb67893346ec978667b3cad2b7fd").then(function(last_reading) {
  console.log(last_reading);
})
```

**Comment by DutchmanNL** [comment]
> As I was just recently looking at the IOBroker GIT ( https://github.com/ioBroker/ ) you got my attention :)

;)

great thank you !

I will start with first try and erros by having it handled within the javascript engine we have.
As soon thats a working scenario i would like to continue with the code blocks to an adapter setup as for enduser its much easier to just install the adapter, provide user and pass and the magic happens (as the other adapters too :)

**Comment by DutchmanNL** [comment]
@zoernert thank you very much, i managed to get the first data by using your example and the API.
Will create a repository on github during the next days and share the link here, and of course i will link you as contributor much appreciate your response !

**Comment by DutchmanNL** [comment]
@zoernert again thank you :)

 just released version 0.2.2 and its working fine : https://github.com/DutchmanNL/ioBroker.discovergy

> Contributors

> AlCalzone

> zoernert
